apps/heureka/src/components/services/ComponentInstancesList.jsx

```diff
@@ -0,0 +1,114 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState } from "react"
+import { useQuery } from "@tanstack/react-query"
+import {
+  ContentHeading,
+  Pagination,
+  Container,
+  Stack,
+  DataGrid,
+  DataGridRow,
+  DataGridCell,
+  DataGridHeadCell,
+} from "@cloudoperators/juno-ui-components"
+import { useGlobalsQueryClientFnReady, useGlobalsQueryOptions, useGlobalsActions } from "../StoreProvider"
+import LoadElement from "../shared/LoadElement"
+import { severityString, highestSeverity } from "../shared/Helper"
+
+const ComponentInstancesList = ({ serviceCcrn }) => {
+  const queryOptions = useGlobalsQueryOptions("ComponentInstancesOfService")
+  const { setQueryOptions } = useGlobalsActions()
+  const queryClientFnReady = useGlobalsQueryClientFnReady()
+  const [currentPage, setCurrentPage] = useState(1)
+
+  const { data, isLoading } = useQuery({
+    queryKey: [
+      "ComponentInstancesOfService",
+      {
+        ...queryOptions,
+        filter: { serviceCcrn: [serviceCcrn] },
+      },
+    ],
+    enabled: !!queryClientFnReady && !!serviceCcrn,
+  })
+
+  const items = data?.ComponentInstances?.edges || []
+  const pageInfo = data?.ComponentInstances?.pageInfo || {}
+  const totalPages = pageInfo?.pages?.length || 0
+
+  const onPaginationChanged = (newPage) => {
```

Comment:
I would recommend to extract this logic to an own component including the Pagination component since you might probably use the pagination in each list and it would give you also the change to tested easier. What do you think?